libtenzir/builtins/operators/delay.cpp

```diff
@@ -137,12 +138,36 @@ class delay_operator final : public crtp_operator<delay_operator> {
   std::optional<time> start_ = {};
 };
 
-class plugin final : public virtual operator_plugin<delay_operator> {
+class plugin final : public virtual operator_plugin<delay_operator>,
+                     public virtual operator_factory_plugin {
 public:
   auto signature() const -> operator_signature override {
     return {.source = true};
   }
 
+  auto make(invocation inv, session ctx) const
+    -> failure_or<operator_ptr> override {
+    // XXX: Perhaps update delay_operator to deal with located<> instead if the
+    // older parser supports it?
+    auto speed = std::optional<located<double>>{};
+    auto start = std::optional<time>{};
+    auto field = std::string{};
```

Comment:
This should probably be an expression instead for TQL2 that must evaluate to an array of timestamps. Might make sense to pair with @jachris for a quick intro for this.